tests/test_activate.py

```diff
@@ -1,74 +1,194 @@
+from conda.cli.activate import pathlist_to_str
+import subprocess
+import tempfile
+
+# make pathsep unicode for sake of windows backslash string formatting
+pathsep = u"%s" % pathsep
+
+def path_identity(path):
+    """Used as a dummy path converter where no conversion necessary"""
+    return path
+
+# defaults for unix shells.  Note: missing "exe" entry, which should be set to
+#    either an executable on PATH, or a full path to an executable for a shell
+unix_shell_base = dict(ps_var="PS1",
+                       echo="echo",
+                       test_echo_extra="",
+                       var_format="${var}",
+                       binpath="/bin/",  # mind the trailing slash.
+                       source_setup="source",
+                       nul='2>/dev/null',
+                       set_var='export ',
+                       shell_suffix="",
+                       printps1='echo $PS1',
+                       printdefaultenv='echo $CONDA_DEFAULT_ENV',
+                       printpath="echo $PATH",
+                       raw_ps=os.getenv("PS1", ""),
+                       shell_args="-l -c",
+                       path_from=path_identity,
+                       path_to=path_identity,
+                       slash_convert=(u"\\", u"/"),
+)
+
+if platform.startswith("win"):
+    shells = {
+        #"powershell": dict(
+        #    echo="echo",
+        #    test_echo_extra=" .",
+        #    ps_var="PS1",
+        #    var_format="${var}",
+        #    binpath="/bin/",  # mind the trailing slash.
+        #    source_setup="source",
+        #    nul='2>/dev/null',
+        #    set_var='export ',
+        #    shell_suffix=".ps",
+        #    printps1='echo $PS1',
+        #    printdefaultenv='echo $CONDA_DEFAULT_ENV',
+        #    printpath="echo %PATH%",
+        #    raw_ps=os.getenv("PROMPT", ""),
+        #    exe="powershell.exe",
+        #    path_from=path_identity,
+        #    path_to=path_identity,
+        #    slash_convert = (u"/", u"\\"),
+        #),
+        "cmd.exe": dict(
+            echo="echo",
+            ps_var="PROMPT",
+            var_format="%{}%",
+            binpath="\\Scripts\\",  # mind the trailing slash.
+            source_setup="call",
+            test_echo_extra="",
+            nul='1>NUL 2>&1',
+            set_var='set ',
+            shell_suffix=".bat",
+            printps1="echo %PROMPT%",
+            printdefaultenv='IF NOT "%CONDA_DEFAULT_ENV%" == "" (\n'
+                            'echo %CONDA_DEFAULT_ENV% ) ELSE (\n'
+                            'echo()', # parens mismatched intentionally.  See http://stackoverflow.com/questions/20691060/how-do-i-echo-a-blank-empty-line-to-the-console-from-a-windows-batch-file
+            printpath="echo %PATH%",
+            raw_ps=os.getenv("PROMPT", ""),
+            exe="cmd.exe",
+            shell_args="/d /c",
```

Comment:
As you are later using `shell_args.split(" ")`, it might be better to store a list instead here: `shell_args=['/d', '/c']`. This prevents possible errors in the future of the kind `shell_args='/a /b /c "C:\\Program Files"'`...